examples/BasicExample/ESP32BasicExample.ino

```diff
@@ -0,0 +1,133 @@
+#include <WiFi.h>
+#include <SocketIoClient.h>
+
+/////////////////////////////////////
+////// USER DEFINED VARIABLES //////
+///////////////////////////////////
+/// WIFI Settings ///
+const char* ssid     = "yourSSID";
+const char* password = "yourWiFiPassword";
+
+/// Socket.IO Settings ///
+char host[] = "192.168.1.2"; // Socket.IO Server Address
+int port = 5000; // Socket.IO Port Address
+char path[] = "/socket.io/?transport=websocket"; // Socket.IO Base Path
+bool useSSL = false; // Use SSL Authentication
+const char * sslFingerprint = "";  // SSL Certificate Fingerprint
+bool useAuth = false; // use Socket.IO Authentication
+const char * serverUsername = "socketIOUsername";
+const char * serverPassword = "socketIOPassword";
+
+/// Pin Settings ///
+int LEDPin = 2;
+int buttonPin = 0;
+
+
+/////////////////////////////////////
+////// ESP32 Socket.IO Client //////
+///////////////////////////////////
+
+SocketIoClient webSocket;
+WiFiClient client;
+
+bool LEDState = false;
+
+
+void socket_Connected(const char * payload, size_t length) {
+  Serial.println("Socket.IO Connected!");
+}
+
+void socket_statusCheck(const char * payload, size_t length) {
+  char* message = "\"OFF\"";
+  if (!LEDState) {
+    message = "\"ON\"";
+  }
+  webSocket.emit("status", message);
+}
+
+void socket_event(const char * payload, size_t length) {
+  Serial.print("got message: ");
+  Serial.println(payload);
+}
+
+void socket_pushButton(const char * payload, size_t length) {
+  LEDStateChange(!LEDState);
+}
+
+void LEDStateChange(const bool newState) {
+  char* message = "\"OFF\"";
+  if (!newState) {
+    message = "\"ON\"";
+  } 
+  webSocket.emit("state_change", message);
+  LEDState = newState;
+  Serial.print("LED state has changed: ");
+  Serial.println(message);
+}
+
+void checkLEDState() {
+  if (LEDState) {
```

Comment:
You can simplify this to
```suggestion
  digitalWrite(LEDPin, LEDState);
```

Reply:
Yup, was copy pasta-ing from a client that I built and stripped this down. Good catch.